include/zephyr/pm/policy.h

```diff
@@ -250,26 +250,6 @@ void pm_policy_device_power_lock_get(const struct device *dev);
 void pm_policy_device_power_lock_put(const struct device *dev);
 
 #ifdef CONFIG_PM_LOW_LATENCY_EVENTS
-/**
- * @brief Register a low latency event.
- *
- * This is same as pm_policy_event_register but adds a low latency property
- * to the event. The system will make sure to be up when the event occurs.
- * For more precision, it takes as argument the absolute time in cycles
- *
- * @note It is mandatory to unregister events once they have happened by using
- * pm_policy_event_unregister(). Not doing so is an API contract violation,
- * because the system would continue to consider them as valid events in the
- * *far* future, that is, after the cycle counter rollover.
- *
- * @param evt Event.
- * @param cycle When the event will occur, in absolute time.
- *
- * @see pm_policy_event_unregister
- */
-void pm_policy_low_latency_event_register(struct pm_policy_event *evt,
-					  uint32_t cycle);
```

Comment:
What happened to this function? Why is it removed?

Reply:
@hubertmis in my third commit I replaced the relative time used by pm_policy_event_register and you can see that the two functions : pm_policy_low_latency_event_register and pm_policy_event_register are almost identical.
That's why I removed it.
Do you think for user it is better to have a separate function for low latency events even if the code is almost the same ?

Reply:
Do I understand correctly that all events are now considered low latency? Is it intentional? It would result in increased power consumption caused by early wake-ups before "any-latency" events.

Reply:
@hubertmis No actually all the events are considered low_latency only if this option is enabled : CONFIG_PM_LOW_LATENCY_EVENTS
Maybe for more clarification I can keep the pm_policy_low_latency_event_register with a low_latency argument to be able to register events that (do or do not) require to be handled as low_latency.
What do you think ?

Reply:
I think changing a function's properties (`pm_policy_event_register()`) based on Kconfig (`PM_LOW_LATENCY_EVENTS`) is a confusing design. In a library, I could use `pm_policy_event_register()` expecting to handle events with low latency, but then the application could force latency on my library by changing Kconfig settings.

It should be clear from the API user's perspective if they register a low-latency or any-latency event. To make it clear I see the following options:

1. Make `pm_policy_low_latency_event_register()` register only low-latency events. This function returns an error or is unavailable if `PM_LOW_LATENCY_EVENTS` is disabled
2. Add a parameter `bool low_latency` to `pm_policy_low_latency_event_register()`, function returns an error if `low_latency` is `true` and `PM_LOW_LATENCY_EVENTS` is disabled
3. Keep two functions, the low latency one available only when `PM_LOW_LATENCY_EVENTS` is enabled.

I would say that option 3 is the cleanest API.

Reply:
I think option 3 is better, keeping both functions.
I have in mind as well to add the argument low_latency to pm_policy_low_latency_event_register to set the initial state of the event (low_latency set to true or false) this property could be updated as well using pm_policy_low_latency_event_update